otelconf/v0.3.0/propagator_test.go

```diff
@@ -0,0 +1,195 @@
+			wantErr: false,
+		},
+		{
+			name: "valid tracecontext",
+			cfg: configOptions{
+				opentelemetryConfig: OpenTelemetryConfiguration{
+					Propagator: &Propagator{
+						Composite: []*string{ptr("tracecontext")},
+					},
+				},
+			},
+			want:    propagation.TraceContext{},
+			wantErr: false,
+		},
+		{
+			name: "valid baggage",
+			cfg: configOptions{
+				opentelemetryConfig: OpenTelemetryConfiguration{
+					Propagator: &Propagator{
+						Composite: []*string{ptr("baggage")},
+					},
+				},
+			},
+			want:    propagation.Baggage{},
+			wantErr: false,
+		},
+		{
+			name: "valid b3",
+			cfg: configOptions{
+				opentelemetryConfig: OpenTelemetryConfiguration{
+					Propagator: &Propagator{
+						Composite: []*string{ptr("b3")},
+					},
+				},
+			},
+			want:    b3.New(),
+			wantErr: false,
+		},
+		{
+			name: "valid b3multi",
+			cfg: configOptions{
+				opentelemetryConfig: OpenTelemetryConfiguration{
+					Propagator: &Propagator{
+						Composite: []*string{ptr("b3multi")},
+					},
+				},
+			},
+			want:    b3.New(b3.WithInjectEncoding(b3.B3MultipleHeader)),
+			wantErr: false,
+		},
+		{
+			name: "valid jaeger",
+			cfg: configOptions{
+				opentelemetryConfig: OpenTelemetryConfiguration{
+					Propagator: &Propagator{
+						Composite: []*string{ptr("jaeger")},
+					},
+				},
+			},
+			want:    jaeger.Jaeger{},
+			wantErr: false,
+		},
+		{
+			name: "valid xray",
+			cfg: configOptions{
+				opentelemetryConfig: OpenTelemetryConfiguration{
+					Propagator: &Propagator{
+						Composite: []*string{ptr("xray")},
+					},
+				},
+			},
+			want:    xray.Propagator{},
+			wantErr: false,
+		},
+		{
+			name: "valid ottrace",
+			cfg: configOptions{
+				opentelemetryConfig: OpenTelemetryConfiguration{
+					Propagator: &Propagator{
+						Composite: []*string{ptr("ottrace")},
+					},
+				},
+			},
+			want:    ot.OT{},
+			wantErr: false,
+		},
+		{
+			name: "multiple propagators",
+			cfg: configOptions{
+				opentelemetryConfig: OpenTelemetryConfiguration{
+					Propagator: &Propagator{
+						Composite: []*string{ptr("tracecontext"), ptr("baggage"), ptr("b3")},
+					},
+				},
+			},
+			want:    propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{}, b3.New()),
+			wantErr: false,
+		},
+		{
+			name: "empty composite",
+			cfg: configOptions{
+				opentelemetryConfig: OpenTelemetryConfiguration{
+					Propagator: &Propagator{
+						Composite: []*string{},
+					},
+				},
+			},
+			want:    propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{}),
+			wantErr: false,
+		},
+		{
+			name: "empty propagator name",
+			cfg: configOptions{
+				opentelemetryConfig: OpenTelemetryConfiguration{
+					Propagator: &Propagator{
+						Composite: []*string{ptr(""), ptr("tracecontext")},
+					},
+				},
+			},
+			want:    propagation.TraceContext{},
+			wantErr: false,
+		},
```

Comment:
Wouldn't it mean that somebody explicitly set an empty string? Is not an error scenario? Shouldn't we failed the parsing if anything is wrong in the YAML file?